simulation/sub8_gazebo/nodes/fake_perception.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/python
+from geometry_msgs.msg import Pose, Point, PoseStamped, Vector3
+from mil_misc_tools import text_effects
+from mil_msgs.srv import SetGeometry
+import numpy as np
+import rospy
+from std_srvs.srv import SetBool, SetBoolResponse
+from sub8_msgs.srv import VisionRequestResponse, VisionRequest, VisionRequest2D
+from gazebo_msgs.srv import GetModelState
+from std_msgs.msg import Header
+import os
+import rospkg
+import sys
+import yaml
+
+
+rospack = rospkg.RosPack()
+config_file = os.path.join(rospack.get_path(
+    'sub8_missions'), 'sub8', 'vision_proxies.yaml')
+f = yaml.load(open(config_file, 'r'))
+
+fprint = text_effects.FprintFactory(title="SIMULATOR").fprint
+
+
+def handle_fake_perception(extra, target_object):
+    '''
+    Calls the GetModelState service from gazebo to get the realtime position of the model targeted.
+    Provides this information to the mission.
+    @param extra The target_name passed through some missions.
+    Other missions do not pass a target_name thus its lable of extra.
+    @param target_object Is the model name of the object targeted by the mission.
+    Missions that do not pass a target_name must use this.
+    '''
+
+    now = rospy.get_rostime()
+    k = np.uint32(0)
+    if extra != '':
+        target_object = extra
+    if target_object == '':
+        fprint("NO TARGET")
+        sys.exit(0)
+    model = get_position(target_object)
+    pose_stamp = PoseStamped(header=Header(seq=k, stamp=now, frame_id="/map"),
+                             # Offset our pose by the starting position of the sub relative to the world in Gazebo.
+                             pose=Pose(position=Point(model.pose.position.x - 13, model.pose.position.y - 24, -1),
+                                       orientation=model.pose.orientation))
+    covariance_diagonal = Vector3(0, 0, 0)
+    found = True
+    resp2 = VisionRequestResponse(pose_stamp, covariance_diagonal, found)
+    return resp2
+
+
+def get_position(model_name):
+
+    rospy.wait_for_service('/gazebo/get_model_state')
```

Comment:
Best to make this wait for service happen in the main function and the service proxy to be declared in main as a global. This will make this function (which is run a lot) to finish a lot faster